src/ui/screen.rs

```diff
@@ -188,182 +162,159 @@ impl History {
-    output_line: u16,
-    prompt_line: u16,
     status_area: StatusArea,
     cursor_prompt_pos: u16,
     history: History,
     scroll_data: ScrollData,
+    title: String,
+    prompt_text: String,
 }
 
 impl Screen {
     pub fn new() -> Result<Self, Box<dyn error::Error>> {
-        let screen = AlternateScreen::from(stdout().into_raw_mode()?);
+        let backend = TermionBackend::new(AlternateScreen::from(stdout().into_raw_mode()?));
+        let terminal = Terminal::new(backend)?;
         let (width, height) = termion::terminal_size()?;
 
         let status_area_height = 1;
-        let output_line = height - status_area_height - 1;
-        let prompt_line = height;
 
-        let status_area = StatusArea::new(status_area_height, output_line + 1, width);
+        let status_area = StatusArea::new(status_area_height);
 
         Ok(Self {
-            screen,
+            terminal,
             width,
             height,
-            output_line,
             status_area,
-            prompt_line,
-            cursor_prompt_pos: 1,
+            cursor_prompt_pos: 0,
             history: History::new(),
             scroll_data: ScrollData(false, 0),
+            title: "".to_string(),
+            prompt_text: "".to_string(),
         })
     }
 
+    /// Initialize the screen
     pub fn setup(&mut self) -> Result<()> {
-        self.reset()?;
-        write!(self.screen, "{}", termion::clear::All)?;
+        self.clear()?;
 
         // Get params in case screen resized
-        let (width, height) = termion::terminal_size()?;
-        if width > 0 && height > 0 {
-            self.width = width;
-            self.height = height;
-            self.output_line = height - self.status_area.height() - 1;
-            self.prompt_line = height;
-
-            write!(
-                self.screen,
-                "{}{}",
-                ScrollRegion(OUTPUT_START_LINE, self.output_line),
-                DisableOriginMode
-            )
-            .unwrap(); // Set scroll region, non origin mode
-            self.redraw_top_bar("", 0)?;
-            self.reset_scroll()?;
-            self.screen.flush()?;
+        let size = self.terminal.size()?;
+        if size.width > 0 && size.height > 0 {
+            self.width = size.width;
+            self.height = size.height;
+
+            self.reset_scroll();
+            self.goto_prompt()?;
             Ok(())
         } else {
             Err(TerminalSizeError.into())
         }
     }
 
-    pub fn set_status_area_height(&mut self, height: u16) -> Result<()> {
-        let height = height.max(1).min(5);
-        self.status_area.set_height(height, self.height);
-        self.setup()?;
-        Ok(())
-    }
+    pub fn redraw(&mut self) -> Result<()> {
+        let main_window = {
+            let block = Block::default()
+                .borders(Borders::TOP)
+                .border_style(Style::default().fg(Color::Green))
+                .title(self.title.as_str());
```

Comment:
Is it possible to indent the title as it was before? (prepended by one 'bar-char')

Reply:
I changed the border type to double, as it was before. I also prepended TWO bar-chars.
For some reason, if I only prepend one, it ignores it... I looked into the tui code and couldn't see a reason as to why that should be, but that's what it did.
Putting two bar-chars renders both... Don't really get it, but I also don't think it matters overly much.